File: schemaorgphp/classes/payment_card.php
<?php

class PaymentCard extends FinancialProduct implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'PaymentCard';
	
	/**
	 * A cardholder benefit that pays the cardholder a small percentage of their net expenditures.
	 * see : https://pending.schema.org/cashBack
	 * @var boolean|boolean[]|float|float[]
	 */
	public var $cash_back;
	
	/**
	 * A secure method for consumers to purchase products or services via debit, credit or smartcards by using RFID or NFC technology.
	 * see : https://pending.schema.org/contactlessPayment
	 * @var boolean|boolean[]
	 */
	public var $contactless_payment;
	
	/**
	 * A floor limit is the amount of money above which credit card transactions must be authorized.
	 * see : https://pending.schema.org/floorLimit
	 * @var \MonetaryAmount|\MonetaryAmount[]
	 */
	public var $floor_limit;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'PaymentCard'
		);
		
		$serialized = so_json_serialize( $this->cash_back );
		if ( ! empty( $serialized ) ) {
			$out['cashBack'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->contactless_payment );
		if ( ! empty( $serialized ) ) {
			$out['contactlessPayment'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->floor_limit );
		if ( ! empty( $serialized ) ) {
			$out['floorLimit'] = $serialized;
		}
		
		return $out;
	}
}

File: schemaorggo/payment_card.go
package schemaorggo

import "encoding/json"

// PaymentCard see : https://schema.org/PaymentCard
type PaymentCard struct {
	FinancialProduct

	typeContext

	// CashBack see : https://pending.schema.org/cashBack
	// A cardholder benefit that pays the cardholder a small percentage of their net expenditures.
	// types : Boolean Number
	CashBack []interface{} `json:"cashBack,omitempty"`

	// ContactlessPayment see : https://pending.schema.org/contactlessPayment
	// A secure method for consumers to purchase products or services via debit, credit or smartcards by using RFID or NFC technology.
	// types : Boolean
	ContactlessPayment []bool `json:"contactlessPayment,omitempty"`

	// FloorLimit see : https://pending.schema.org/floorLimit
	// A floor limit is the amount of money above which credit card transactions must be authorized.
	// types : MonetaryAmount
	FloorLimit []*MonetaryAmount `json:"floorLimit,omitempty"`
}

func (v PaymentCard) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.FinancialProduct.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.CashBack
		if len(v.CashBack) == 1 {
			value = v.CashBack[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["cashBack"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.ContactlessPayment
		if len(v.ContactlessPayment) == 1 {
			value = v.ContactlessPayment[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["contactlessPayment"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.FloorLimit
		if len(v.FloorLimit) == 1 {
			value = v.FloorLimit[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["floorLimit"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v PaymentCard) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "PaymentCard"

	return data, nil
}

func (v PaymentCard) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}
